mod/cp_notifications/start.php

```diff
@@ -499,21 +506,30 @@ function cp_overwrite_notification_hook($hook, $type, $value, $params) {
 
 	$subject = htmlspecialchars_decode($subject,ENT_QUOTES);
 
-	foreach ($to_recipients as $to_recipient) {
-		// username for link in footer (both email notification and site notification
-		$message['user_name'] = $to_recipient->username;
-		if ($cp_msg_type != 'cp_event_ics') {
-			$template = elgg_view('cp_notifications/email_template', $message);
-			$site_template = elgg_view('cp_notifications/site_template', $message);
+	if (is_array($to_recipients)) {
+		foreach ($to_recipients as $to_recipient) {
+			// username for link in footer (both email notification and site notification
+			$message['user_name'] = $to_recipient->username;
+			if ($cp_msg_type != 'cp_event_ics') {
+				$template = elgg_view('cp_notifications/email_template', $message);
+				$site_template = elgg_view('cp_notifications/site_template', $message);
+			}
+
+			$newsletter_appropriate = array('cp_wire_share','cp_messageboard','cp_wire_mention','cp_hjpost','cp_hjtopic', 'cp_friend_request', 'cp_friend_approve');
+			if (strcmp(elgg_get_plugin_user_setting('cpn_set_digest', $to_recipient->guid,'cp_notifications'),'set_digest_yes') == 0 && in_array($cp_msg_type, $newsletter_appropriate)) {
+				$result = create_digest($author, $cp_msg_type, $content_entity, $to_recipient, $content_url);
+				continue;
+
+			} else
+				$result = (elgg_is_active_plugin('phpmailer')) ? phpmailer_send( $to_recipient->email, $to_recipient->name, $subject, $template ) : mail($to_recipient->email, $subject, $template, cp_get_headers($event));
+
+			messages_send($subject, $site_template, $to_recipient->guid, $sender_guid, 0, true, $add_to_sent);
 		}
+	}
 
-		$newsletter_appropriate = array('cp_wire_share','cp_messageboard','cp_wire_mention','cp_hjpost','cp_hjtopic', 'cp_friend_request', 'cp_friend_approve');
-		if (strcmp(elgg_get_plugin_user_setting('cpn_set_digest', $to_recipient->guid,'cp_notifications'),'set_digest_yes') == 0 && in_array($cp_msg_type, $newsletter_appropriate)) {
-			$result = create_digest($author, $cp_msg_type, $content_entity, $to_recipient, $content_url);
-			continue;
-		} else
-			$result = (elgg_is_active_plugin('phpmailer')) ? phpmailer_send( $to_recipient->email, $to_recipient->name, $subject, $template ) : mail($to_recipient->email, $subject, $template, cp_get_headers($event));
-		messages_send($subject, $site_template, $to_recipient->guid, $sender_guid, 0, true, $add_to_sent);
+		// register the error, if either of the arrays are not populated
+	if (!is_array($to_recipients)) {
```

Comment:
This could just be the `else` of the previous conditional or was that not working correctly

Reply:
fixed the indent for the comment